rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -1701,6 +1719,129 @@ rocalVideoFileSource(
 }
 
 RocalTensor ROCAL_API_CALL
+rocalNumpyFileSource(
+    RocalContext p_context,
+    const char* source_path,
+    unsigned internal_shard_count,
+    bool is_output,
+    bool shuffle,
+    bool loop,
+    std::pair<RocalLastBatchPolicy, bool> last_batch_info) {
+    Tensor* output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    try {
+        auto max_dimensions = evaluate_numpy_data_set(StorageType::NUMPY_DATA, DecoderType::SKIP_DECODE,
+                                                      source_path);
+
+        RocalTensorlayout tensor_format = RocalTensorlayout::NONE;
+        RocalTensorDataType tensor_data_type;
+        std::unordered_map<int, RocalTensorDataType> data_type_map = {
+            {0, RocalTensorDataType::FP32},
+            {1, RocalTensorDataType::FP16},
+            {2, RocalTensorDataType::UINT8},
+            {3, RocalTensorDataType::INT8},
+            {4, RocalTensorDataType::UINT32},
+            {5, RocalTensorDataType::INT32},
+        };
+        auto dtype = max_dimensions.at(max_dimensions.size() - 1);
+        max_dimensions.pop_back();
+        tensor_data_type = data_type_map[dtype];
+        unsigned num_of_dims = max_dimensions.size() + 1;
+        std::vector<size_t> dims;
+        dims.resize(num_of_dims);
+        dims[0] = context->user_batch_size();
+        for (uint i = 0; i < max_dimensions.size(); i++)
+            dims[i + 1] = max_dimensions[i];
```

Comment:
std::copy for vector won't work?

Reply:
All TensorInfo constructors used move semantics for passing dims in this file